boranga/frontend/boranga/src/components/internal/species_communities/dashboard.vue

```diff
@@ -46,23 +67,84 @@ export default {
         FormSection,
     },
     computed: {
-        is_flora: function(){
+        /*------properties to show the user authenticated Tabs only-----------*/
+        showFloraTab: function(){
+            return this.group_types.includes('flora');
+        },
+        showFaunaTab: function(){
+            return this.group_types.includes('fauna');
+        },
+        showCommunityTab: function(){
+            return this.group_types.includes('community');
+        },
+        /*---------------------------------------------------------------------*/
+        /*---------properties to load group related vue components-------------*/
+        isFlora: function(){
             return this.filterGroupType == 'flora';
         },
-        is_fauna: function(){
+        isFauna: function(){
             return this.filterGroupType == 'fauna';
         },
-        is_community: function(){
+        isCommunity: function(){
             return this.filterGroupType == 'community';
         },
+        /*---------------------------------------------------------------------*/
+    },
+    methods: {
+        set_tabs: function(){
+            let vm = this;
+            /* set user preference tab by default on load of dashboard (Note: doesn't affect on the load group component)*/
+            if(vm.user_preference === 'flora'){
+                $('#pills-tab a[href="#pills-flora"]').tab('show');
+            }
+            if(vm.user_preference === 'fauna'){
+                $('#pills-tab a[href="#pills-fauna"]').tab('show');
+            }
+            if(vm.user_preference === 'community'){
+                $('#pills-tab a[href="#pills-community"]').tab('show');
+            }
+        },
+        load_group_datatable: function(grouptype){
+            /*----------to set the  filterGroupType to load the particular component only----------*/
+            if(grouptype === 'flora'){
+                this.filterGroupType = 'flora';
+            }
+            else if(grouptype === 'fauna'){
+                this.filterGroupType = 'fauna';
+            }
+            else if(grouptype === 'community'){
+                this.filterGroupType = 'community';
```

Comment:
Can we just say:
this.filterGroupType = grouptype?

Reply:
Yes I agree. :)